lib/facility_builder.rb

```diff
@@ -0,0 +1,52 @@
+class FacilityBuilder
+  def build_facility_data(dmv_data)
+    facilities = []
+    dmv_data.each do |dmv_entry|
+      
+      
+      entry_name = nil
+      entry_address = nil
+      entry_phone = nil
+
+      if dmv_entry.key?(:dmv_office)
+        entry_name = dmv_entry[:dmv_office].strip
+        entry_address = format_address(dmv_entry, 1)
+        entry_phone = dmv_entry[:phone]
+      elsif dmv_entry.key?(:office_name)
+        entry_name = dmv_entry[:office_name].strip
+        entry_address = format_address(dmv_entry, 2)
+        entry_phone = dmv_entry[:public_phone_number]
+      elsif dmv_entry.key?(:name)
+        entry_name = dmv_entry[:name].strip
+        entry_address = format_address(dmv_entry, 3)
+        entry_phone = dmv_entry[:phone]
+      end
+
+      new_facility = Facility.new(name: entry_name, address: entry_address, phone: entry_phone)
+      
+      format_facility_services(new_facility, dmv_entry)
+      facilities << new_facility      
+    end
+    return facilities
+  end
+
+  def format_address(dmv_entry, source_num)
+    if source_num == 1
```

Comment:
I would probably set this up so that instead of getting a number as the source, we would get a state code like "CO".  That would be a little more readable, especially if we end up needing to deal with all 50 states' data.